lib/json-schema/validator.rb

```diff
@@ -474,6 +474,8 @@ def merge_missing_values(source, destination)
 
         if @@json_backend == 'yajl'
           @@serializer = lambda{|o| Yajl::Encoder.encode(o) }
+        elsif @@json_backend == 'json'
+          @@serializer = lambda{|o| JSON.dump(o) }
```

Comment:
Why change the serialisation method? So far as I'm aware (as I could be wrong) it's only used to generate a cache key for a given schema. I assume its performance related? If so I wonder if we could change to using Hash#hash rather than serializing the whole schema out to a string and generating the key from that.